scipy/sparse/tests/test_base.py

```diff
@@ -1653,22 +1683,42 @@ def test_pow(self):
         A = array([[1, 0, 2, 0], [0, 3, 4, 0], [0, 5, 0, 0], [0, 6, 7, 8]])
         B = self.spcreator(A)
 
-        for exponent in [0,1,2,3]:
-            ret_sp = B**exponent
-            ret_np = np.linalg.matrix_power(A, exponent)
-            assert_array_equal(ret_sp.toarray(), ret_np)
-            assert_equal(ret_sp.dtype, ret_np.dtype)
-
-        # invalid exponents
-        for exponent in [-1, 2.2, 1 + 3j]:
-            assert_raises(ValueError, B.__pow__, exponent)
-
-        # nonsquare matrix
-        B = self.spcreator(A[:3,:])
-        assert_raises(TypeError, B.__pow__, 1)
+        if self.is_array_test:
+            # Todo: Add test of B**(1+3j) when np1.24 is no longer supported.
+            #    Complex exponents of 0 changed for numpy-1.25
+            #    from `(nan+nanj)` to `0`. Old value makes our result dense
+            #    and is hard to check for without any `isnan` method.
+            # So, while untested here, complex exponents work with np>=1.25.
+            for exponent in [1, 2, 2.2, 3]:
+                ret_sp = B**exponent
```

Comment:
Double checking, but this seems to be checking element-wise exponent, instead of matrix exponent?